core/src/main/java/dev/streamx/cli/ingestion/publish/payload/PayloadResolver.java

```diff
@@ -131,49 +107,17 @@ private void replaceValues(DocumentContext documentContext, List<ValueArguments>
     }
   }
 
-  private static JsonNode extractReplacement(ValueArguments valueArgument,
-      Pair<JsonPath, String> extract) {
-    JsonPath jsonPath = extract.getKey();
-    String value = extract.getValue();
-
-    if (valueArgument.isBinary()) {
-      byte[] bytes = readContent(value);
-
-      return JacksonUtils.toJsonNode(bytes);
-    } else if (valueArgument.isString()) {
-      String content = readStringContent(value);
-
-      return TextNode.valueOf(content);
-    } else {
-      return prepareWrappedJsonNode(
-          value,
-          (exception, source) -> {
-            throw ValueException.jsonParseException(exception, jsonPath, source);
-          },
-          (exception, source) -> {
-            throw ValueException.genericJsonProcessingException(exception, jsonPath, source);
-          }
-      ).json();
+  private JsonNode extractPayloadFragment(PayloadArgument payloadArgument, String value)
+      throws IOException {
+    RawPayload rawPayload = sourceResolver.resolve(value);
+    SourceType sourceType = SourceType.STRING;
+    if (payloadArgument.isBinary()) {
```

Comment:
SourceType could be returned directly from PayloadArgument. It would simplify the contract.